### PR TITLE
Add MR images and pixel_representation flag

### PR DESCRIPTION
**Pull request by thinks**
@rickardraysearch 

Verified created images with dciodvfy.exe.

### CODE REVIEW

**Comment by thinks** [comment]
@rickardholmberg @rickardraysearch 

Should be fine to merge now!

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ dicomutils
 
 A set of utilities for working with DICOM files.
 
-The main utility is currently build_dicom, which can generate simple synthetic CT data, 
+The main utility is currently build_dicom, which can generate simple synthetic CT data, MR data,
```
Comment (rickardraysearch) [inline README.md]: PET?

File: build_dicom.py [hunk 2]
```diff
@@ -32,22 +45,33 @@ def __call__(self, parser, namespace, values, option_string=None):
...
                     help='The nominal energy of beams in an RT Plan.')
 parser.add_argument('--values', dest='values', default=[], action='append', metavar='VALUE | SHAPE{,PARAMETERS}',
                     help="""Set the Hounsfield or dose values in a volume to the given value.\n\n\n
                     For syntax, see the forthcoming documentation or the source code...""")
-parser.add_argument('--center', dest='center', default="[0;0;0]", help="""Center of the image, in dicom patient coordinates.""")
+parser.add_argument('--pixel-representation', dest='pixel_representation', default='signed',
+                    choices=['signed', 'unsigned'],
+                    help='signed: Stored pixel value type is int16, unsigned: Stored pixel value type is uint16.')
+parser.add_argument('--rescale-slope', dest='rescale_slope', type=float,
+                    help="""Set the rescale slope (defaults - CT: 1.0, MR: 1.0).""")
```
Comment (rickardraysearch) [inline build_dicom.py]: PET?

File: build_dicom.py [hunk 3]
```diff
@@ -85,15 +116,65 @@ def __call__(self, parser, namespace, values, option_string=None):
...
+                center=np.array(series.center))
+        elif series.modality == "PT":
+            if 'PatientPosition' not in sb.current_study:
+                parser.error("Patient position must be specified when writing MR images!")
+
+            ib = sb.build_pt(
+                num_voxels=num_voxels,
+                voxel_size=voxel_size,
+                pixel_representation=pixel_representation,
+                rescale_slope=rescale_slope,
```
Comment (rickardraysearch) [inline build_dicom.py]: rescale_slope seems undefined?

File: build_dicom.py [hunk 4]
```diff
@@ -138,6 +219,8 @@ def __call__(self, parser, namespace, values, option_string=None):
             sb.current_study['PatientsBirthDate'] = series.patients_birthdate
         if series.modality == "CT":
             ib.build()
+        elif series.modality == "MR":
+            ib.build()
         elif series.modality == "RTDOSE":
```
Comment (rickardraysearch) [inline build_dicom.py]: PET too?

File: builders.py [hunk 5]
```diff
@@ -89,11 +89,38 @@ def __init__(self, patient_position="HFS", patient_id="", patients_name = "", pa
...
+                      rescale_slope=rescale_slope,
+                      rescale_intercept=rescale_intercept,
+                      column_direction=column_direction,
+                      row_direction=row_direction,
+                      slice_direction=slice_direction)
         self.seriesbuilders['CT'].append(b)
         return b
 
+    def build_mr(self, num_voxels, voxel_size, pixel_representation, center=None, column_direction=None,
+                 row_direction=None, slice_direction=None):
```
Comment (rickardraysearch) [inline builders.py:106]: rescale_slope & rescale_intercept?

File: builders.py [hunk 6]
```diff
@@ -89,11 +89,38 @@ def __init__(self, patient_position="HFS", patient_id="", patients_name = "", pa
...
+                      center=center,
+                      column_direction=column_direction,
+                      row_direction=row_direction,
+                      slice_direction=slice_direction)
+        self.seriesbuilders['MR'].append(b)
+        return b
+
+    def build_pt(self, num_voxels, voxel_size, pixel_representation, center=None, column_direction=None,
+                 row_direction=None, slice_direction=None):
+        b = PTBuilder(self.current_study, num_voxels, voxel_size, pixel_representation,
```
Comment (rickardraysearch) [inline builders.py]: rescale_slope?

File: builders.py [hunk 7]
```diff
@@ -179,17 +225,140 @@ def real_value_to_stored_value(self, real_value):
...
+
+class MRBuilder(ImageBuilder):
+    def __init__(
+            self,
+            current_study,
+            num_voxels,
+            voxel_size,
+            pixel_representation,
+            center=None,
+            column_direction=None,
```
Comment (rickardraysearch) [inline builders.py:257]: rescale_slope & rescale_intercept?

File: builders.py [hunk 8]
```diff
@@ -179,17 +225,140 @@ def real_value_to_stored_value(self, real_value):
...
+
+
+class PTBuilder(ImageBuilder):
+    def __init__(
+            self,
+            current_study,
+            num_voxels,
+            voxel_size,
+            pixel_representation,
+            center=None,
```
Comment (rickardraysearch) [inline builders.py:314]: rescale_slope?

File: README.md (hunk 1 above)
Comment (thinks) [reply README.md]: Fixed.

File: build_dicom.py (hunk 2 above)
Comment (thinks) [reply build_dicom.py]: Fixed

Comment (thinks) [reply build_dicom.py]: Also, removed slope/icept for MR so removed from here too.

File: builders.py (hunk 5 above)
Comment (thinks) [reply builders.py:106]: slope/icept not applicable for MR, it was a mistake to add it and it was later removed.

File: builders.py (hunk 6 above)
Comment (thinks) [reply builders.py]: Added

File: build_dicom.py (hunk 3 above)
Comment (thinks) [reply build_dicom.py]: Fixed

File: build_dicom.py (hunk 4 above)
Comment (thinks) [reply build_dicom.py]: Fixed

File: builders.py (hunk 8 above)
Comment (thinks) [reply builders.py:314]: Fixed

File: builders.py (hunk 7 above)
Comment (thinks) [reply builders.py:257]: Not applicable for MR

**Comment by thinks** [comment]
I think going forward it is risky to use default arguments in "backend" code, better to handle that kind of stuff in build_dicom.py. Otherwise there is a risk that the values are not passed down correctly without generating errors...

**Comment by thinks** [comment]
ping @rickardraysearch

**Comment by rickardraysearch** [comment]
Unexpectedly, I was allowed to push to this branch. Sorry for that - the last commit 3d8488c115addc28e622c4c0e4d032e45fc5c376 was only intended to be pull request comments... Other than that, it looks good!

**Comment by thinks** [comment]
Ok, feel free to merge (I don't have write access).

**Pull request merged**